src/NHibernate/Proxy/ProxyBuilderHelper.cs

```diff
@@ -134,25 +136,48 @@ internal static MethodBuilder GetObjectDataMethodBuilder(TypeBuilder typeBuilder
 			return methodBuilder;
 		}
 
+		private static readonly System.Type[] _equalsParametersTypes = { typeof(object) };
+
 		internal static MethodBuilder GenerateMethodSignature(string name, MethodInfo method, TypeBuilder typeBuilder)
 		{
-			//TODO: Should we use attributes of base method?
-			var methodAttributes = MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.Virtual;
+			var explicitImplementation = method.DeclaringType.IsInterface;
+			if (explicitImplementation &&
+				(typeBuilder.BaseType == typeof(object) ||
+#pragma warning disable 618
+					typeBuilder.BaseType == typeof(ProxyDummy)) &&
```

Comment:
Old dynamic proxy legacy. Not handling this causes its `InterfaceWithEqualsGethashcodeTests` test to fail.